### PR TITLE
fix(cli-preview): merge process.env when starting services

### PR DESCRIPTION
**Pull request by kuojianlu**
merge process.env when starting services, otherwise there will be an error on macos: `env: node: No such file or directory` or `/bin/sh: dotnet: command not found`

### CODE REVIEW

File: packages/cli/src/cmds/preview/commonUtils.ts [hunk 1]
```diff
@@ -158,3 +158,20 @@ export async function getPortsInUse(
...
+
+export function mergeProcessEnv(
+  env: { [key: string]: string | undefined } | undefined
+): { [key: string]: string | undefined } | undefined {
+  const processEnv = process.env;
+  if (env === undefined) {
+    return process.env;
+  }
+  const keys = Object.keys(env);
+  for (const key of Object.keys(processEnv)) {
```
Comment (swatDong) [inline packages/cli/src/cmds/preview/commonUtils.ts:170]: How about `Object.assign()`?

``` typescript
const result = Object.assign({}, process.env);
for (const key of Object.keys(env)) {
    result[key] = env[key];
}
```

File: packages/cli/src/cmds/preview/commonUtils.ts [hunk 2]
```diff
@@ -158,3 +158,20 @@ export async function getPortsInUse(
...
+    const existing = keys.some((k) => k === key);
+    if (!existing) {
+      env[key] = processEnv[key];
```
Comment (swatDong) [inline packages/cli/src/cmds/preview/commonUtils.ts:173]: Is it expected to modify input env?

File: packages/cli/src/cmds/preview/commonUtils.ts (hunk 1 above)
Comment (kuojianlu) [reply packages/cli/src/cmds/preview/commonUtils.ts]: Nice code. Updated.

File: packages/cli/src/cmds/preview/commonUtils.ts (hunk 2 above)
Comment (kuojianlu) [reply packages/cli/src/cmds/preview/commonUtils.ts]: Safe here, but it is not a good practice to modify input argument. Updated.

**Review by swatDong — APPROVED** [review]

**Pull request merged**